### PR TITLE
Fix notification race condition and failed error catching

### PR DESCRIPTION
**Pull request by thiswillbeyourgithub**
Bugfixes form [this thread](https://github.com/wasp-os/wasp-os/issues/222)

- fix error 'a bytes-like object is required, not 'str''
- fix: race condition for a notification
- docs: mention in the tools argument that verbosity can be further increased

### CODE REVIEW

File: nrf_ble_dfu_controller.py [hunk 1]
```diff
@@ -198,10 +204,24 @@ def _dfu_wait_for_notify(self):
                 #
                 self.ble_conn.sendline('')
                 string = self.ble_conn.before
-                if '[   ]' in string:
+                if b'[   ]' in string:
                     print('Connection lost! ')
                     raise Exception('Connection Lost')
-                return None
+                 else:
```
Comment (agittins) [inline nrf_ble_dfu_controller.py]: This else comment has an extra space of indent.
```suggestion
                else:
```

**Review by agittins — APPROVED** [review]
This certainly resolved the error on my system and resulted in successfully flashing my pinetime.

Only caveat being the extra space of indent on line 210.

**Comment by CMoH** [comment]
I had to checkout this PR to be able to get unstuck on my pinetime, after flashing `reloader-mcuboot.zip`. The only issue was the extra space - easily fixable.

**Comment by thiswillbeyourgithub** [comment]
@agittins  omg I'm so sorry I missed your message so I hadn't approved of this change. Thank you very much.

@CMoH thanks to you I finally saw it :)

Pinging @daniel-thompson

**Pull request merged**